### DISCUSSION TITLE
getting delegate failed while working with mp4 files

### DISCUSSION THREAD

**Discussion by EarsKilla** [dlemstra/Magick.NET#826] [Help]
Don't have any ideas how to fix this:
```
delegate failed `"ffmpeg.exe" -nostdin -v -1 -i "%M%%d.pam" -plays %I "%u.%m"' @ error/delegate.c/InvokeDelegate/1898
```

Tried:
- put `ffmpeg.exe` around my app exe (`...\bin\Debug\netcoreapp3.1`)
- put folder with `ffmpeg.exe` into path
- downgrade from latest `7.22.2.2` to `7.22.0.0` because of [this](https://github.com/ImageMagick/ImageMagick/issues/2720)
- setting different platform targeting (Any CPU (current) / 32-bit / 64-bit)
- different builds of ffmpeg (x86 and x64)

Packages related to Magick.NET installed:
- Magick.NET-Q8-AnyCPU 7.22.2.2
- Magick.NET.Core 5.2.2

Environment:
- Windows 10 x64 1809
- ```cmd
  >dotnet --list-sdks
  2.1.807 [C:\Program Files\dotnet\sdk]
  3.1.301 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]
  ```
- targeting .net-core 3.1

---

On linux machine (debian 10) it's failing with another error:
```
NoTagFound `video:decode' @ error/delegate.c/InvokeDelegate/1736
```
```bash
$ dotnet --list-sdks
2.1.811 [/usr/share/dotnet/sdk]
2.2.402 [/usr/share/dotnet/sdk]
3.0.103 [/usr/share/dotnet/sdk]
3.1.201 [/usr/share/dotnet/sdk]
3.1.404 [/usr/share/dotnet/sdk]
5.0.100 [/usr/share/dotnet/sdk]
```
ffmpeg version 4.1.6-1~deb10u1

Note: Project was rebuilded on linux machine

Another note: planning to develop on windows but run on linux

**Comment by EarsKilla** [comment]
here is code:
```cs
        private async Task<MemoryStream> DoAnimatedMeme(Stream data)
        {
            using var imageCollection = new MagickImageCollection(data, MagickFormat.Mp4);

            var framesCount = imageCollection.Count;
            for (var i = 0; i < framesCount; i++)
            {
                var frame = imageCollection[i];

                var oldHeight = frame.Height;
                var oldWidth = frame.Width;

                var percentage = new Percentage(100 - CalcPercentage(i + 1, framesCount, 70));
                frame.LiquidRescale(percentage);

                frame.Resize(oldWidth, oldHeight);
            }

            var mem = new MemoryStream();
            await imageCollection.WriteAsync(mem).ConfigureAwait(false);

            mem.Position = 0;

            return mem;
        }
```

PS: need to work with streams because downloading file stream from the internet, process this file and upload back

**Comment by dlemstra** [comment]
When running on Linux you will need to use the ffmpeg executable that was compiled on that platform. Starting with the next version `ffmpeg` will be executed instead of `ffmpeg.exe`.
